meson.build

```diff
@@ -33,25 +33,41 @@ libnvme_dep = dependency('libnvme', fallback : ['libnvme', 'libnvme_dep'])
 
 # Check for libuuid availability
 libuuid = dependency('uuid', required: true)
-conf.set('CONFIG_LIBUUID', libuuid.found(), description: 'Is libuuid required?')
+conf.set('CONFIG_LIBUUID', libuuid.found(), description: 'Is libuuid available?')
 
 # Check for libjson-c availability
 json_c = dependency('json-c', version: '>=0.13', fallback : ['json-c', 'json_c_dep'])
 if json_c.found()
-  conf.set('CONFIG_JSONC', true, description: 'Is json-c required?')
+  conf.set('CONFIG_JSONC', true, description: 'Is json-c available?')
   conf.set('LIBJSONC_14', json_c.version().version_compare('>=0.14'), description: 'Is json-c at least 0.14?')
 endif
 
-# Check for libhugetlbfs  availability
-libhugetlbfs = dependency('hugetlbfs', required: false)
-conf.set('LIBHUGETLBFS', libhugetlbfs.found(), description: 'Is libhugetlbfs required?')
+# Check for libhugetlbfs availability
+if cc.has_header(['hugetlbfs.h'])
+  libhugetlbfs_dep = cc.find_library('hugetlbfs',
+                                     required : false)
+  have_libhugetlbfs = libhugetlbfs_dep.found()
+else
+  libhugetlbfs_dep = []
+  have_libhugetlbfs = false
+endif
+conf.set('LIBHUGETLBFS', have_libhugetlbfs, description: 'Is libhugetlbfs available?')
 
 # Check for zlib availability
-libz_dep = dependency('zlib', required: true)
-
-# Check for open-ssl availability
-libopenssl = dependency('openssl', version: '>=1.1.0', required: false)
-conf.set('OPENSSL', libopenssl.found(), description: 'Is open-ssl required?')
+libz_dep = dependency('zlib', fallback : ['zlib', 'zlib_dep'])
+
+# Check for OpenSSL availability
+want_openssl = get_option('openssl')
+if want_openssl != 'false'
```

Comment:
I would have suggested using a Feature option: https://mesonbuild.com/Build-options.html#features

You can simply pass a feature option directly to the `required: ` kwarg and it automatically handles:
- disabled? return a not-found dependency, and log to the effect of "skipped dependency openssl because feature openssl is disabled"
- enabled? This is equivalent to your code for "true"
- auto? This is equivalent to your code for "auto"


The neat thing is that by doing this you get:
- better informational logging
- a few less lines, no need to if/else anything and openssl_dep.found() returns `false` when the option is disabled.

Reply:
Thanks for the tip! I suspected that I was doing something wrong. The Feature option is very handy! 

See #1291

Reply:
Great, happy to help. ;) That change LGTM.